crates/uv/src/commands/tool/install.rs

```diff
@@ -380,7 +381,52 @@ pub(crate) async fn install(
             &cache,
             printer,
         )
-        .await?;
+        .await;
+
+        let resolution = match resolution {
+            Ok(resolution) => resolution,
+            Err(
+                err @ ProjectError::Operation(pip::operations::Error::Resolve(
+                    uv_resolver::ResolveError::NoSolution(_),
+                )),
+            ) => {
+                if python_request.is_some() {
+                    return Err(err.into());
+                }
+
+                let interpreter = PythonInstallation::find_or_download(
+                    Some(&PythonRequest::parse("3.12")),
```

Comment:
I'm looking for the best way to derive the necessary Python version error from a `[NoSolutionError](https://github.com/astral-sh/uv/blob/1602b5c8d7c63ef175e0fe1bb5088c6e9f8f26b1/crates/uv-resolver/src/error.rs#L108)` instance (potentially via a combination of an explicit call to fmt/regex?). Will keep digging, but open to thoughts on how best to approach this.

Reply:
Python requirements are tracked as incompatibilities in the resolver https://github.com/astral-sh/uv/blob/fe88c108136f689c54bc5a9c15df472433d36312/crates/uv-resolver/src/resolver/mod.rs#L2223-L2234 so you should have a `External::FromDependencyOf` with a `PubGrubPackageInner::Python` package in the derivation tree. I'm not sure this is the best method, but should be feasible.